examples/main/main.cpp

```diff
@@ -31,20 +31,21 @@
 #pragma warning(disable: 4244 4267) // possible loss of data
 #endif
 
-static llama_context           ** g_ctx;
-static llama_model             ** g_model;
-static gpt_params               * g_params;
-static std::vector<llama_token> * g_input_tokens;
-static std::ostringstream       * g_output_ss;
-static std::vector<llama_token> * g_output_tokens;
+static llama_context                 ** g_ctx;
+static llama_model                   ** g_model;
+static gpt_params                     * g_params;
+static std::vector<llama_token>       * g_input_tokens;
+static std::ostringstream             * g_output_ss;
+static std::vector<llama_token>       * g_output_tokens;
+static std::vector<llama_chat_msg>    * g_chat_msgs;
```

Comment:
No need for the chat messages to be in the global state - here we add just the stuff that is needed by the `sigint_handler`. Instead, modify `chat_add_and_format` to accept a reference to `chat_msgs`

Reply:
Done in https://github.com/ggerganov/llama.cpp/pull/8068/commits/43cab6bfc6b61896857077eecde818499db7da8a